service/history/consts/const.go

```diff
@@ -68,6 +68,8 @@ var (
 	ErrActivityTaskNotCancelRequested = serviceerror.NewInvalidArgument("unable to mark activity as canceled without activity being request canceled first")
 	// ErrWorkflowCompleted is the error to indicate workflow execution already completed
 	ErrWorkflowCompleted = serviceerror.NewNotFound("workflow execution already completed")
+	// ErrWorkflowZombie is the error to indicate workflow execution is in zombie state and cannot be updated
+	ErrWorkflowZombie = serviceerror.NewNotFound("workflow execution is in zombie state cannot be updated")
```

Comment:
You can reuse `ErrStaleReference` here and use `fmt.Errorf` to provide details.

Reply:
make sense. Updated.